test/Microsoft.AspNetCore.Blazor.Build.Test/ComponentRenderingRazorIntegrationTest.cs

```diff
@@ -2,101 +2,319 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Linq;
-using System.Reflection;
-using Microsoft.AspNetCore.Blazor.Components;
-using Microsoft.AspNetCore.Blazor.Layouts;
 using Microsoft.AspNetCore.Blazor.RenderTree;
 using Microsoft.AspNetCore.Blazor.Test.Helpers;
+using Microsoft.CodeAnalysis.CSharp;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Blazor.Build.Test
 {
     public class ComponentRenderingRazorIntegrationTest : RazorIntegrationTestBase
     {
+        internal override bool UseTwoPhaseCompilation => true;
+
         [Fact]
-        public void SupportsChildComponentsViaTemporarySyntax()
+        public void Render_ChildComponent_Simple()
         {
-            // Arrange/Act
-            var testComponentTypeName = FullTypeName<TestComponent>();
-            var component = CompileToComponent($"<c:{testComponentTypeName} />");
+            // Arrange
+            AdditionalSyntaxTrees.Add(CSharpSyntaxTree.ParseText(@"
+using Microsoft.AspNetCore.Blazor.Components;
+
+namespace Test
+{
+    public class MyComponent : BlazorComponent
```

Comment:
Just to check, these tests that use `BlazorComponent`, they'd also work if it inherited `IComponent` directly would they? Or does it depend on something in the `BlazorComponent` base class?

Reply:
Yeah, I just did this brevity. The tests focused on component discovery include `IComponent` as well.